src/Microsoft.OData.Client/DeleteLinkUriOption.cs

```diff
@@ -0,0 +1,26 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DeleteLinkUriOption.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Client
+{
+    /// <summary>
+    /// Used to specify the form of Uri to be used for a delete link request.
+    /// </summary>
+    public enum DeleteLinkUriOption
+    {
+        /// <summary>
+        /// Pass the id of the related entity as a query param, i.e.,
+        /// {ServiceUri}/{EntitySet}/{Key}/{NavigationProperty}/$ref?$id={ServiceUri}/{RelatedEntitySet}/{RelatedKey}
+        /// </summary>
+        DollarIdQueryParam = 0,

```

Comment:
I'd suggest the name to be `IdQueryParam`. In case we have situtations where the `$` is optional.

Reply:
`$` sign is only optional in system query options e.g select, top, expand etc